artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQConsumer.java

```diff
@@ -493,4 +495,10 @@ private boolean isRolledBack(MessageReference messageReference) {
       }
       return rollbackedMessageRefs.contains(messageReference);
    }
+
+   /** For tests only */
+   public Set<MessageReference> listRolledbackMessageRefs() {
```

Comment:
Why not just make the existing `org.apache.activemq.artemis.core.protocol.openwire.amq.AMQConsumer#getRolledbackMessageRefs` public?

Reply:
Also perhaps rather than make it public at all, make it package-private or protected and add a test-tree class in the same package as it to act as an accessor, e.g Clebert uses this approach, for an example (which just changes such a class) see https://github.com/apache/activemq-artemis/commit/2383aa0125320713b9a753668b203a878c24b2e0#diff-7426cf013ea833e8f1c3cd71cd26e9b584755e13d1fc4ccb4bc95025598795f6

Reply:
Nice method!

Reply:
Robbie's idea is even better.

Reply:
Okay... I will try to get that working. Do I have to do anything in particular to be able to run the test classes from the integration-tests-suite? Might be an issue with my setup but I can't import: `org.apache.activemq.artemis.core.protocol.openwire.amq.AMQConsumerTest `
straight up?

Reply:
@jbertram @gemmellr @franz1981 Is this looking better? Have not seen or done that kind of access before but it seems to be working like you described...

Reply:
Not entirely no. I'll admit I hadnt noticed the test was in a different module when I gave that example where they are in the same one, but I still wouldnt approach it in this way.

Add the new test 'accessor' method in its own new class in the same package, not to the actual test class such as AMQConsumerTest, and place the new class near to the test that needs to use it, i.e in the integration tests module in this case. That will obviously be a split-package, but the integration-tests are not published and arent moduled so that shouldnt cause any issue currently, and theres then no need to add the far uglier test-jar usage.

Reply:
@gemmellr Like this?

Reply:
Yep, thats better